src/watcher.js

```diff
@@ -331,14 +331,25 @@ Watcher.prototype.teardown = function () {
  * @param {*} val
  */
 
-function traverse (val) {
+function traverse (val, walkedObjs) {
   var i, keys
+
+  !walkedObjs && (walkedObjs = {})
```

Comment:
Uh, it's just personal habits, there is no difference between them, do not be feeling trouble with it.